apps/ftu/js/ui.js

```diff
@@ -123,6 +123,26 @@ var UIManager = {
       this[toCamelCase(name)] = document.getElementById(name);
     }.bind(this));
 
+    // Setup settings observers
+    this._settingsObserveHandler = {
+      'geolocation.enabled': {
+        defaultValue: this.geolocationCheckbox.checked,
+        callback: function(value) {
+          if (this.geolocationCheckbox.checked != value) {
```

Comment:
oh, I assumed that .checked could only be true/false, and didn't think about non boolean possibilities on `value`.
We could use `!!this.geolocationCheckbox.checked !== !!value` then to make both boolean, but it seems a little weird (to say the least), and we would end up with same results as using `!=` 
In this case, go with your favourite, but please leave a comment for the future us :smile: